### PR TITLE
Use catch2 instead of ctest

### PR DESCRIPTION
**Pull request by Aakash-kaushik**
ctest reports tests to pass even if they are failed, this is done just so we know if some tests fail. 

reference comment: https://github.com/mlpack/models/pull/63#issuecomment-870899105

### CODE REVIEW

**Comment by Aakash-kaushik** [comment]
How does this even happen, please ignore this PR for now, i didn't mean to include all the resnet changes.

**Comment by birm** [comment]
Do you need guidance on fixing your branch?

**Comment by Aakash-kaushik** [comment]
> Do you need guidance on fixing your branch?

I would highly appreciate it.

**Comment by zoq** [comment]
Looks like you created a branch from the resnet branch and not from the master branch?

File: .ci/linux-steps.yaml [hunk 1]
```diff
@@ -49,7 +49,7 @@ steps:
   displayName: 'Build models'
 
 # Run CTests.
-- script: cd build/tests/ && sudo CTEST_OUTPUT_ON_FAILURE=1 ctest -T Test .
+- script: cd build/ && ./models_test -x 50
```
Comment (zoq) [inline .ci/linux-steps.yaml:52]: I think it's `bin/models_test`, you can add a `cd build/ && ls && bin/models_test -x` to list the files in the directory.

**Comment by Aakash-kaushik** [comment]
> Looks like you created a branch from the resnet branch and not from the master branch?

This was a bit more messier than that, if it's fine i would open another PR by tomorrow or just fix this one up, i mean just opening another one would be pretty easy to me

File: .ci/linux-steps.yaml (hunk 1 above)
Comment (Aakash-kaushik) [reply .ci/linux-steps.yaml:52]: Yup yup it is that, and i did that with this commit: https://github.com/mlpack/models/pull/69/commits/ea0f8165f49375225f0c4263fdecdbc8713309e2 but again i messed this up a bit so tomorrow morning would be a better option.

**Comment by zoq** [comment]
Sure, whatever is easier for you.

**Pull request closed without merging**